### PR TITLE
Fix warning on setting main class

### PR DESCRIPTION
**Pull request by Goooler**
## Checklist

- [ ] PR description added
- [ ] tests are added
- [ ] `CHANGELOG.md` is updated

In case of adding a new rule:
- [ ] `README.md` is updated
- [ ] Rule has been applied on Ktlint itself and violations are fixed

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -54,7 +54,7 @@ dependencies {
 task ktlint(type: JavaExec, group: LifecycleBasePlugin.VERIFICATION_GROUP) {
   description = "Check Kotlin code style including experimental rules."
   classpath = configurations.ktlint
-  main = 'com.pinterest.ktlint.Main'
+  mainClass.set("com.pinterest.ktlint.Main")
```
Comment (Goooler) [inline build.gradle:57]: Fix #1184.

File: .github/workflows/gradle-snapshot-build.yml [hunk 2]
```diff
@@ -15,17 +19,8 @@ jobs:
...
           java-version: 8
+      - uses: gradle/gradle-build-action@v2
       - name: Publish snapshot to Maven
-        uses: gradle/gradle-build-action@v2
-        with:
-          arguments: clean publishMavenPublicationToMavenCentralRepository --no-daemon --no-parallel
-        env:
-          SONATYPE_NEXUS_USERNAME: ${{ secrets.SONATYPE_NEXUS_USERNAME }}
-          SONATYPE_NEXUS_PASSWORD: ${{ secrets.SONATYPE_NEXUS_PASSWORD }}
+        run: ./gradlew clean publishMavenPublicationToMavenCentralRepository --no-daemon --no-parallel
```
Comment (Goooler) [inline .github/workflows/gradle-snapshot-build.yml:24]: Simplify changes in #1398.

File: .github/workflows/gradle-pr-build.yml [hunk 3]
```diff
@@ -9,18 +9,17 @@ jobs:
     strategy:
       matrix:
         os: [ ubuntu-latest, windows-latest ]
+        java: [ 8, 11, 17 ]
```
Comment (Goooler) [inline .github/workflows/gradle-pr-build.yml:12]: Follow up #1416.

**Comment by shashachu** [comment]
Thanks for the fix! I'll merge this just as soon as I finish the 0.45.0 deploy.

**Pull request merged**